### ISSUE TITLE
It is not respecting the padding id in the tokenizer

### ISSUE THREAD

**Issue by bratao** [vespa-engine/vespa#34300]
**Describe the bug**

I'm trying to use a custom cross-encoder reranker and got terrible results. After a lot of debugging, I narrowed it down to the padding ID it uses.
In this model, the tokenizer has the following configuration:
"padding": {
    "strategy": "BatchLongest",
    "direction": "Right",
    "pad_to_multiple_of": null,
    "pad_id": 50283,
    "pad_type_id": 0,
    "pad_token": "[PAD]"
  },
But when debugging, Vespa sets the padding token as "0". There is the customTokenInputIdsfunction, but it only allows setting the start_sequence_id and sep_sequence_id.
What can I do to make the Vespa tokenizer respect the correct padding token or any workaround to replace the "0" to the correct id for the reranker?

**To Reproduce**
1. Just set up a ranker that use a cross-encoder that have a different Padding ID.

**Expected behavior**
It output the correct padding id.

**Vespa version**
8.533.16

**Additional context**

**Comment by thomasht86** [comment]
Hi @bratao - are you able to provide a model and tokenizer, so that we can test? 
Are you on https://slack.vespa.ai/ ? If so, we can take it by DM?

**Comment by bratao** [comment]
@thomasht86 yes I´m. Lets talk there. But an reranker with a different PADDING such as https://huggingface.co/Alibaba-NLP/gte-reranker-modernbert-base would be degraded because of this.

**Comment by thomasht86** [comment]
Posting answer here as well for posterity. 
Just tested this tokenizer and model, and it seem to work correctly with this configuration:

`services.xml`: 
```txt
<?xml version="1.0" encoding="UTF-8"?>
<services version="1.0">
    <container id="reranking_container" version="1.0">
        <search></search>
        <document-api></document-api>
        <document-processing></document-processing>
        <component id="tokenizer" type="hugging-face-tokenizer">
            <model url="https://huggingface.co/Alibaba-NLP/gte-reranker-modernbert-base/resolve/main/tokenizer.json"/>
        </component>
    </container>
    <content id="reranking_content" version="1.0">
        <redundancy>1</redundancy>
        <documents>
            <document type="doc" mode="index"></document>
        </documents>
        <nodes>
            <node distribution-key="0" hostalias="node1"></node>
        </nodes>
    </content>
</services>
```

`schema.sd`:
```txt
schema doc {
    document doc {
        field id type string {
            indexing: summary | attribute
        }
        field text type string {
            indexing: index | summary
            index: enable-bm25
        }
    }
    field body_tokens type tensor<float>(d0[512]) {
        indexing: input text | embed tokenizer | attribute | summary
    }
    fieldset default {
        fields: text
    }
    onnx-model crossencoder {
        file: files/crossencoder.onnx
        input input_ids: input_ids
        input attention_mask: attention_mask
        output logits: logits
    }
    rank-profile bm25 {
        first-phase {
            expression {
                bm25(text)
            }
        }
    }
    rank-profile reranking inherits default {
        inputs {
            query(q) tensor<float>(d0[64])         }
        function input_ids() {
            expression {
                customTokenInputIds(1, 2, 512, query(q), attribute(body_tokens))
            }
        }
        function attention_mask() {
            expression {
                tokenAttentionMask(512, query(q), attribute(body_tokens))
            }
        }
        first-phase {
            expression {
                bm25(text)
            }
        }
        global-phase {
            expression {
                sigmoid(onnx(crossencoder).logits{d0:0,d1:0})
            }
            rerank-count: 10
        }
        summary-features {
            query(q)
            input_ids
            attention_mask
            onnx(crossencoder).logits
        }
    }
}
```

It also works by only replacing the model and tokenizer url in [this](https://vespa-engine.github.io/pyvespa/examples/cross-encoders-for-global-reranking.html) pyvespa notebook. 
Let us know it solves your issue.

**Comment by thomasht86** [comment]
Key here is probably to use [huggingface-tokenizer-embedder](https://docs.vespa.ai/en/reference/embedding-reference.html#huggingface-tokenizer-embedder) and not [huggingface-embedder](https://docs.vespa.ai/en/reference/embedding-reference.html#huggingface-embedder), as you can only "embed" the tokens, and can't run inference of the model until you have both query and document tokens.
